app/src/main/java/com/github/wnder/user/FirebaseUserDatabase.java

```diff
@@ -86,8 +87,13 @@ public FirebaseUserDatabase(Context context){
         CompletableFuture<String> cf = new CompletableFuture<>();
         // Get all pictures and remove the ones not in radius
         getAllIdsAndLocation().thenAccept( idsAndLocation -> {
-            Location userLocation = user.getPositionFromGPS((LocationManager)context.getSystemService(Context.LOCATION_SERVICE),context);
 
+            // Remove skipped pictures
+            for (String pictureId: user.getSkippedPictures()) {
+                idsAndLocation.remove(pictureId);
+            }
```

Comment:
You could maybe use (I guess):
idsAndLocation.keySet().removeAll(pictureId);

Reply:
I tried it and it didn't seem to work but I might have done something wrong.

Reply:
Weird, I probably missed somthing when trying myself.